docs/enhancements/on-prem-data-collection.md

```diff
@@ -0,0 +1,121 @@
+---
+title: on-prem-data-collection
+authors:
+  - "@CrystalChun"
+creation-date: 2022-09-30
+last-updated: 2022-09-30
+---
+
+# On-premises Data Collection
+
+## Summary
+
+To provide a better installation process and experience, the Assisted Installer ("AI") will collect data about installations when it is deployed on-premises as part of either [Red Hat Advanced Cluster Management for Kubernetes](https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.6) ("RHACM") or as a stand-alone Podman deployment. Data was already collected from clusters deployed on [Red Hat's Hybrid Cloud Console](https://console.redhat.com/openshift/assisted-installer/clusters) (“SaaS”). The collection of data from on-premises deployments will give the team additional insight into the product's usage, customers' pain points, and analytics for how to best improve the product.
+
+## Motivation
+
+Data helps us see how the product is being used, helps us to analyze trends, and to determine ways we can improve the product for our users. 
+
+### Goals
+
+1. Create a process to collect and send [event](https://github.com/openshift/assisted-service/tree/master/docs/dev/events.md) data from on-premises deployments of AI.
+2. Analyze and process on-premises data (by use of visualizations, dashboards, etc.) at Red Hat like SaaS data.
+
+
+### Non-Goals
+
+1. Implementing new data types to collect is out of the scope of this epic. This epic defines how and when the data already collected will be processed and sent.
+2. Define how SaaS should redesign their data pipeline (taken care of in [MGMT-11244](https://issues.redhat.com/browse/MGMT-11244)) is out of the scope of this epic. This epic only refers to on-premises deployments of AI.
+3. Define the creation and/or management of an event streaming instance such as Kafka and the endpoint this data will be hosted at (both taken care of in [MGMT-11244](https://issues.redhat.com/browse/MGMT-11244)).
+4. Collecting must-gather data (and logs) for debugging failed cluster installation is outside the scope of this epic. This epic is only concerned with [event](https://github.com/openshift/assisted-service/blob/6b8401d3b0f0ceea5a837f33c17b9b1cd5ec52a3/internal/events/api/event.go) data.
+
+
+## Proposal
+
+A new flow for sending data is to be added:
+When a cluster installation attempt finishes (regardless of success or failure), all of the event data for this cluster will be queried from the DB, packaged, and sent to the data hosting endpoint.
+
+Changes that need to be made in the `Assisted-Service` to support this:
+- Needs to know what the data hosting service endpoint is and how to authenticate into it
```

Comment:
can be also local s3 / elk, right?

Reply:
Thought it would need to be a remote/cloud s3/elk for us to be able to pull it? @rccrdpccl @tsorya do you know?

Reply:
I was thinking of having a local target that allowed you to specify a directory visible to assisted service, where both events and logs would be kept.

Then, on podman assisted service (like _openshift-install agent_ installations), we'd mount some location to the pod. Continuing with the _openshift-install agent_ case, we'd then, after writing the image to disk in node0, we'd mount node0 and writing it there, so it can be retrieved after installation completes.

Reply:
Interesting, I think I need some clarification. The local directory/mount would be an additional change to Assisted Service, but the service would still need to know the endpoint of the data hosting service, right?

Reply:
isn't that just a different implementation of the client that is going to be used? 
for example send events is the interface and the implementation details can be file/s3/elk etc.
as long as the interface is clear there should be no difference no?

Reply:
That makes sense, thank you for clarifying @filanov! FYI @rccrdpccl it looks like this is also targeted for disconnected environments

Reply:
@celebdor how do envision it in case of billi - what will send it post installation? And even worth, what will send it in case of the installation failure - because this is when we'll need this data, but there wont be any control plane.

Reply:
Isn't the data collection going to be independently managed from the install command trigger? A failure or retries can slow down the start of the installation itself. Also, what happens to the post install events?

Reply:
Did we agree that the only scope of this enhancement is sending data specifically to the SaaS ingress endpoint?

I don't see value in collecting the event data somewhere locally even if we wanted to deploy some separate component to store it it.

Reply:
Agreeing with Nick that there isn't much value collecting the event data locally.